src/scripts/main.js

```diff
@@ -1,3 +1,202 @@
 'use strict';
 
-// write code here
+const trList = Array.from(document.querySelectorAll('tr th')).slice(0, 5);
+const tBody = document.querySelector('tbody');
+const rows = Array.from(tBody.querySelectorAll('tr'));
+
+function parseSalary(salaryString) {
+  return Number(salaryString.replace(/[$,]/g, ''));
+}
+
+const sortState = [];
+
+trList.forEach((headerTh, columnIndex) => {
+  headerTh.addEventListener('click', () => {
+    const ascSort = sortState[columnIndex];
+
+    sortTable(columnIndex, ascSort);
+
+    sortState.fill(false);
+    sortState[columnIndex] = !ascSort;
+  });
+});
+
+function sortTable(columnIndex, ascSort) {
+  const sortedRows = rows.sort((a, b) => {
+    const aText = a.children[columnIndex].textContent.trim();
+    const bText = b.children[columnIndex].textContent.trim();
+
+    const aNum = parseSalary(aText);
+    const bNum = parseSalary(bText);
+
+    if (!isNaN(aNum) && !isNaN(bNum)) {
+      if (ascSort) {
+        return bNum - aNum;
```

Comment:
The sorting logic for numbers seems inverted. When `ascSort` is true, it should return `aNum - bNum` to sort in ascending order, not `bNum - aNum`.